alpha/lib/model/category.php

```diff
@@ -490,28 +490,48 @@ protected function validateFullNameIsUnique()
 		if ($category)
 			throw new kCoreException("Duplicate category: $fullName", kCoreException::DUPLICATE_CATEGORY);
 	}
-	
+
+	private static function shouldMoveEntriesToParentCategory($moveEntriesToParentCategory)
+	{
+		if($moveEntriesToParentCategory === null)
+			return true;
+		if($moveEntriesToParentCategory === 0)
+			return false;
+
+		return true;
+	}
+
+	// instead of deleting a category and its sub-categories recursively this method
+	// will update all the deleted info in the db at once and will iterate through all of the sub-categories and invoke delete actions.
+	//1. get all category childes ids for sub-categories from db in one call
+	//2. update all retrieved category childes ids to be deleted in one call
+	//3. iterate on categories one by one and invoke the delete event
+	//4. delete sub-categories-entries and move all to parent category / also delete sub-categories kusers
+
 	public function setDeletedAt($v, $moveEntriesToParentCategory = null)
 	{
-		if(is_null($this->move_entries_to_parent_category))
-		{
-			if(is_null($moveEntriesToParentCategory))
-				$moveEntriesToParentCategory = $this->getParentId();
-					
-			$this->move_entries_to_parent_category = $moveEntriesToParentCategory;
-			
-			$this->loadChildsForSave();
-			foreach($this->childs_for_save as $child)
-			{
-				$child->setDeletedAt($v, $moveEntriesToParentCategory);
-				$child->save();
-			}
-		}
-		
-		$this->setStatus(CategoryStatus::DELETED);
 		parent::setDeletedAt($v);
+		$this->setStatus(CategoryStatus::DELETED);
+
+		$fullIds = $this->getFullIds();
+		$categoriesIds = self::getCategoriesStartWithIds($fullIds);
+		self::updateDeletedCategoriesInDB($categoriesIds);
+		$categories = self::getCategories($categoriesIds);
+
+		foreach ($categories as $categoryToDelete) {
+				kEventsManager::raiseEvent(new kObjectDeletedEvent($categoryToDelete));
+				kEventsManager::raiseEventDeferred(new kObjectReadyForIndexEvent($categoryToDelete));
+		}
+
+		if ($this->getInheritanceType() == InheritanceType::MANUAL)
+			$this->addDeleteCategoryTreeKuserJob($this->getId(), false, $fullIds);
+		if (self::shouldMoveEntriesToParentCategory($moveEntriesToParentCategory))
```

Comment:
Instead of self::shouldMoveEntriesToParentCategory($moveEntriesToParentCategory), remove the $moveEntriesToParentCategory logic and just check if($this->getParentId()).